package/MDAnalysis/analysis/base.py

```diff
@@ -125,14 +125,105 @@
 import inspect
 import logging
 import itertools
+import warnings
 
 import numpy as np
 from MDAnalysis import coordinates
 from MDAnalysis.core.groups import AtomGroup
 from MDAnalysis.lib.log import ProgressBar
+from typing import Callable, Iterable
+from importlib import import_module
+from functools import partial
 
 logger = logging.getLogger(__name__)
 
+class ParallelExecutor(object):
+    """Class that executes parallel computations for AnalysisBase (but potentially for other purposes)
+    using one of the available backends"""
+
+    @classmethod
+    @property
+    def possible_backends(cls):
```

Comment:
```suggestion
    def available_backends(cls):
```
? Or "supported"? Maybe even just "backends" (and a docstring).

Reply:
I still stand by `available_backends` as more descriptive name

Reply:
Yep. I suggested "availabe" here before seeing you used it elsewhere (later on). What I was objecting is the "possible_backends" you used here.